### PR TITLE
[huggingface_tensorflow, huggingface_pytorch] upgrade Tensorflow to 2.5 & Pytorch to 1.9

### PR DESCRIPTION
**Pull request by philschmid**
*GitHub Issue #, if available:*

#1355

This PR upgrades the PyTorch version from 1.8.1 to 1.9.0 and TensorFlow from 2.4.3 to 2.5.1

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [x] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true` or `neuron_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: dlc_developer_config.toml [hunk 1]
```diff
@@ -32,12 +32,12 @@ ec2_tests = true
 
 ### Off by default
 efa_tests = false
```
Comment (arjkesh) [inline dlc_developer_config.toml:34]: This can remain `false` - there are no EFA tests for HF afaik. The EFA failures we saw in a prior release were related to an unrelated bug that we identified and fixed.

File: huggingface/pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    openjdk-8-jdk-headless \
+    openssl \
+    vim \
+    wget \
+    unzip \
+    zlib1g-dev \
+ && rm -rf /var/lib/apt/lists/* \
+ && apt-get clean
+
+RUN cd /tmp \
```
Comment (arjkesh) [inline huggingface/pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu:83]: possible that temp file test is failing from this?

Comment (philschmid) [reply huggingface/pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu:83]: https://github.com/aws/deep-learning-containers/blob/aa250c6bd2ace66debae750b65298f7af25ef26d/pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu#L80

**Comment by arjkesh** [comment]
Note to reviewer: check that openssl version in HF containers isn't affected by `CVE-2021-3711`

File: huggingface/pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu (hunk 2 above)
Comment (arjkesh) [reply huggingface/pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu:83]: note: we identified the issue was related to apt-get packages. @philschmid  the corresponding [non-HF PT dockerfile](https://github.com/aws/deep-learning-containers/blob/master/pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu#L177-L178) and found the lines removing tmp dir. He has added that to the HF image, waiting for tests to determine whether this resolved the issue.

Comment (arjkesh) [reply huggingface/pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu:83]: Issue appears resolved. Closing this thread

File: huggingface/pytorch/inference/docker/1.9/py3/Dockerfile.cpu [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+ADD https://raw.githubusercontent.com/aws/deep-learning-containers/master/src/deep_learning_container.py /usr/local/bin/deep_learning_container.py
+
+RUN chmod +x /usr/local/bin/deep_learning_container.py
+
+#################################
+# Hugging Face specific section #
+#################################
+
+RUN curl https://aws-dlc-licenses.s3.amazonaws.com/pytorch-1.8/license.txt -o /license.txt
```
Comment (pinaraws) [inline huggingface/pytorch/inference/docker/1.9/py3/Dockerfile.cpu:110]: this should be https://aws-dlc-licenses.s3.amazonaws.com/pytorch-1.9/license.txt

File: huggingface/pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+ADD https://raw.githubusercontent.com/aws/deep-learning-containers/master/src/deep_learning_container.py /usr/local/bin/deep_learning_container.py
+
+RUN chmod +x /usr/local/bin/deep_learning_container.py
+
+#################################
+# Hugging Face specific section #
+#################################
+
+RUN curl https://aws-dlc-licenses.s3.amazonaws.com/pytorch-1.8/license.txt -o /license.txt
```
Comment (pinaraws) [inline huggingface/pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu:156]: this should be https://aws-dlc-licenses.s3.amazonaws.com/pytorch-1.9/license.txt

File: huggingface/tensorflow/inference/docker/2.5/py3/Dockerfile.cpu [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+ADD https://raw.githubusercontent.com/aws/deep-learning-containers/master/src/deep_learning_container.py /usr/local/bin/deep_learning_container.py
+
+RUN chmod +x /usr/local/bin/deep_learning_container.py
+
+#################################
+# Hugging Face specific section #
+#################################
+
+RUN curl https://aws-dlc-licenses.s3.amazonaws.com/tensorflow-2.4/license.txt -o /license.txt
```
Comment (pinaraws) [inline huggingface/tensorflow/inference/docker/2.5/py3/Dockerfile.cpu:110]: this should be https://aws-dlc-licenses.s3.amazonaws.com/tensorflow-2.5/license.txt

File: huggingface/tensorflow/inference/docker/2.5/py3/cu112/Dockerfile.gpu [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+ADD https://raw.githubusercontent.com/aws/deep-learning-containers/master/src/deep_learning_container.py /usr/local/bin/deep_learning_container.py
+
+RUN chmod +x /usr/local/bin/deep_learning_container.py
+
+#################################
+# Hugging Face specific section #
+#################################
+
+RUN curl https://aws-dlc-licenses.s3.amazonaws.com/tensorflow-2.4/license.txt -o /license.txt
```
Comment (pinaraws) [inline huggingface/tensorflow/inference/docker/2.5/py3/cu112/Dockerfile.gpu:105]: this should be https://aws-dlc-licenses.s3.amazonaws.com/tensorflow-2.5/license.txt

Comment (philschmid) [reply huggingface/tensorflow/inference/docker/2.5/py3/cu112/Dockerfile.gpu]: done! thanks for pointing out

File: huggingface/tensorflow/inference/docker/2.5/py3/Dockerfile.cpu (hunk 5 above)
Comment (philschmid) [reply huggingface/tensorflow/inference/docker/2.5/py3/Dockerfile.cpu]: done! thanks for pointing out

File: huggingface/pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu (hunk 4 above)
Comment (philschmid) [reply huggingface/pytorch/inference/docker/1.9/py3/cu111/Dockerfile.gpu]: done! thanks for pointing out

File: huggingface/pytorch/inference/docker/1.9/py3/Dockerfile.cpu (hunk 3 above)
Comment (philschmid) [reply huggingface/pytorch/inference/docker/1.9/py3/Dockerfile.cpu]: done! thanks for pointing out

**Review by pinaraws — APPROVED** [review]

**Review by pinaraws — DISMISSED** [review]

**Review by arjkesh — APPROVED** [review]

**Review by arjkesh — APPROVED** [review]
